lib/toSyllabicSymbols.js

```diff
@@ -0,0 +1,150 @@
+const toUnicode = require( './toUnicode' )
+const { getRegexGroup } = require( './regex-utils' )
+
+// These have no impact on weight, therfore remove before processing
+const zeroWeightSigns = [
+  '੍ਰ',
+  'ੵ',
+  'ੑ',
+  '੍ਵ',
+  '੍ਹ',
+  '੍ਟ',
+  '੍ਨ',
+  '੍ਯ',
+  '੍ਚ',
+  '੍ਤ',
+  '੍',
+  'ਿ',
+  'ੁ',
+  'ਂ',
+  'ਃ',
+  '।',
+  '॥',
+  '☬',
+  '਼',
+  '❁',
+]
+
+// For the rest, need to analyze the string
+// Base characters counts as one syllable (light)
+// Base characters with any number of long (deeragh) sounds counts as two syllables (heavy)
+const syllableSymbols = {
+  light: '1',
+  deeragh: 's',
+  heavy: '2',
+}
+
+// The list of base characters for analysis
+const baseCharacters = [
+  'ਇ',
+  'ਉ',
+  'ਙ',
+  'ੳ',
+  'ਅ',
+  'ਬ',
+  'ਭ',
+  'ਚ',
+  'ਛ',
+  'ਦ',
+  'ਧ',
+  'ੲ',
+  'ਡ',
+  'ਢ',
+  'ਗ',
+  'ਘ',
+  'ਹ',
+  'ਜ',
+  'ਝ',
+  'ਕ',
+  'ਖ',
+  'ਲ',
+  'ਲ਼',
+  'ਮ',
+  'ਨ',
+  'ਪ',
+  'ਫ',
+  'ਤ',
+  'ਥ',
+  'ਰ',
+  'ਸ',
+  'ਸ਼',
+  'ਟ',
+  'ਠ',
+  'ਵ',
+  'ੜ',
+  'ਣ',
+  'ਯ',
+  'ਜ਼',
+  'ਗ਼',
+  'ਖ਼',
+  'ਫ਼',
+  'ਞ',
+]
+
+// The list of long sound characters for analysis
+const deeraghModifiers = [
+  'ੰ',
+  '੍ਹੂ',
+  'ੀ',
+  'ੂ',
+  'ੇ',
+  'ੈ',
+  'ੋ',
+  'ੌ',
+  'ਾਂ',
+  'ਾ',
+  'ੱ',
+]
+
+// Some symbols represent multiple characters in a single unicode entity.
+// These should be considered unsplittable and unmodifiable by other vowels/signs.
+// The following represent a base character and deeragh in one unicode entity point.
+// Since author is unaware of whether these can be further modified with deeragh
+// they'll be processed as a heavy sequence (base char + deeragh) in the mapping.
+// If that is not the case and these characters cannot have further deeragh modifiers,
+// then they can be safely mapped directly to syllableSymbol.heavy
+const twoUnitSyllables = [ 'ਊ', 'ਓ', 'ਈ', 'ਏ', 'ਐ', 'ਆ', 'ਔ' ]
+syllableSymbols.heavySequence = syllableSymbols.light + syllableSymbols.deeragh
```

Comment:
I would prefer this to be in the object definition itself, even if you had to write out the actual string for this

Reply:
If you do, then you'd have to change this on your own in the object if you change light or deeragh

And I was trying to avoid that

Definitely can go in the same object, should I also add a more explicit comment?